src/Build.UnitTests/BackEnd/EventSourceTestHelper.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.Engine.UnitTests.BackEnd
+{
+    internal sealed class EventSourceTestHelper : EventListener
```

Comment:
Please document this class including example or recommended usage.
Also please state that this class is not supposed to be used when tests runs concurrently. Optionally it can implement protection against multi instanciation through a static instance counter or such.
Currently in MSBuild we do not support to run test concurrently but if someone want to reuse this class or we ever start support test runs concurrently it might make tests unstable, as this class might contain tracing events from other concurrent tests as well.

Reply:
Thank you @rokonec. 
Added the summary  with mentioning not having protection against concurently running tests as well. 
However the workaround for concurently running tests is to not to rely on the count of the events, but to the unique generated ids (if and when possible) as it is done in example test.

Reply:
I agree that with current implementation it is not an issue, but theoretically in situation when test A do not expect event with ID nn but thanks to timing test B will produce it it would cause flaky failures.